opendg/data/data.py

```diff
@@ -0,0 +1,328 @@
+            edge_index = torch.transpose(edge_index, 0, 1)
+            event_dict[t] = EventStore(timestamp=t,edges=edge_index,node_feats=None)
+        return event_dict
+
+    def _load_csv(self, file_path: str) -> Dict[str, Any]:
+        r"""Loads a continuous time dynamic graph from a csv file."""
+        os.path.exists(file_path), f"File {file_path} does not exist"
+        edge_dict = {}
+        with open('test_ctdg.csv', mode='r') as ctdg_file:
+            ctdg_reader = csv.reader(ctdg_file)
+            header = next(ctdg_reader)
+            assert len(header) >= 3, "The csv file should have at least 3 columns (source, destination, timestamp)"
+            #* infer the headers
+            header_dict = {header[i]:i for i in range(len(header))}
+            src_idx = header_dict.get('source',None)
+            dst_idx = header_dict.get('destination',None)
+            time_idx = header_dict.get('timestamp',None)
+
+            for edge in ctdg_reader:
+                u = int(edge[src_idx])
+                v = int(edge[dst_idx])
+                t = int(edge[time_idx])
+                if t not in edge_dict:
+                    edge_dict[t] = [(u,v)]
+                else:
+                    edge_dict[t].append((u,v))
+        return edge_dict
+
+
+
+    def __getattr__(self, key: str) -> Any:
+        return getattr(self._store, key)
+    
+    def __setattr__(self, key: str, value: Any):
+        setattr(self._store, key, value)
+
+    def __delattr__(self, key: str):
+        delattr(self._store, key)
+
+    def __getitem__(self, key: str) -> Any:
+        return self._store[key]
+
+    def __setitem__(self, key: str, value: Any):
+        self._store[key] = value
+
+    def __delitem__(self, key: str):
+        if key in self._store:
+            del self._store[key]
+
+    def __copy__(self):
+        return CTDG(self._store.copy())
+
+    def __deepcopy__(self, memo):
+        return CTDG(self._store.copy())
+
+    def __repr__(self) -> str:
+        return f"CTDG({self._store.keys()})"
+    
+    ###########################################################################
+
+    def aggregate_graph(self, start_time:int=None, end_time:int=None) -> torch.Tensor:
+        r"""Aggregates the graph between start_time and end_time, inclusive.
+        Args:
+            start_time (int, optional): The start time of the aggregation. (default: :obj:`None`)
+            end_time (int, optional): The end time of the aggregation. (default: :obj:`None`)
+        Returns:
+            :class:`torch.Tensor`: The aggregated edge index tensor."""
+        if (start_time is None):
+            start_time = self.min_time
+        if (end_time is None):
+            end_time = self.max_time
+        assert start_time <= end_time, "start_time should be less or equal to end_time"
+        assert isinstance(start_time, int) and isinstance(end_time, int), "start_time and end_time should be integers"

```

Comment:
Since they aren't being used to index anything, why do `start_time` and  `end_time` need to be integers? As long as they are numeric, this should be fine?